docs/ceph-csi-upgrade.md

```diff
@@ -0,0 +1,334 @@
+recommended that you use [official
+releases](https://github.com/ceph/ceph-csi/releases) of Ceph-csi. Unreleased
+versions from the master branch are subject to changes and incompatibilities
+that will not be supported in the official releases. Builds from the master
+branch can have functionality changed and even removed at any time without
+compatibility support and without prior notice.
+
+### Upgrading CephFS
+
+upgrading cephfs csi includes upgrade of cephfs driver and as well as
+kubernetes sidecar containers and also the permissions required for the
+kubernetes sidecar containers, lets upgrade the things one by one
+
+git checkout latest release v2.0.0 branch
+
+```bash
+[$] git clone git@github.com:ceph/ceph-csi.git
+[$] git check v2.0.0
+```
+
+*Note* if you are using ceph-csi with kubernetes v1.13 use templates from v1.13
+directory
+
+#### 1. Upgrade CephFS Provisioner resources
+
+upgrade provisioner resources include updating the provisioner RBAC and
+Provisioner deployment
+
+##### 1.1 Update the CephFS Provisioner RBAC
+
+```bash
+[$] kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-provisioner-rbac.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+serviceaccount/cephfs-csi-provisioner configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-external-provisioner-runner configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-external-provisioner-runner-rules configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrolebinding.rbac.authorization.k8s.io/cephfs-csi-provisioner-role configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+role.rbac.authorization.k8s.io/cephfs-external-provisioner-cfg configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+rolebinding.rbac.authorization.k8s.io/cephfs-csi-provisioner-role-cfg configured
+```
+
+**Note:** Ignore the warning messages
+
+##### 1.2 Update the CephFS Provisioner deployment/statefulset
+
+```bash
+[$]kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-cephfsplugin-provisioner.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+service/csi-cephfsplugin-provisioner configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+deployment.apps/csi-cephfsplugin-provisioner configured
+```
+
+**Note:** Ignore the warning messages
+
+wait for the deployment to complete
+
+```bash
+$]kubectl get deployment
```

Comment:
(nit) Some shell commands begin with `[$]` and some with `$]` can retain one of the alternatives across the doc.